include/boost/url/impl/url_base.ipp

```diff
@@ -1400,6 +1400,68 @@ resolve(
 //
 //------------------------------------------------
 
+template <class Charset>
+void
+url_base::
+normalize_octets_impl(
+    int id,
+    Charset const& allowed,
+    op_t& op) noexcept
+{
+    char* it = s_ + u_.offset(id);
+    char* end = s_ + u_.offset(id + 1);
+    char buf = 0;
+    char* dest = it;
+    while (it < end)
+    {
+        if (*it != '%')
+        {
+            *dest = *it;
+            ++it;
+            ++dest;
+            continue;
+        }
+        if (end - it < 3)
+            break;
+
+        // decode unreserved octets
+        detail::decode_unchecked(
```

Comment:
can't we like make a small function `decode_one` that does this, and uses efficient parameters (not string view)

Reply:
Done :)